text/0000-new-maintainer-group-process.md

```diff
@@ -0,0 +1,22 @@
+
+## Implementation
+
+The following are the 3 scenarios in which a new maintainer group would be formed and the procedures for forming each group type.
+
+### New Maintainer Group for New Repositories
+When proposing a new project that the proposer feels does not fall under the domain of an existing maintainer group they can suggest creating a new maintainer group. As part of the proposal, members of the initial maintainer group should be suggested or a request for volunteers should be open by the proposer of the new project. This group will then be approved by the Steering Committee to be the initial set of maintainers.
+
+### New Maintainer Group for Existing Repositories that are Maintained by One Existing Group
+When the repositories exist and are already under the control of an existing maintainer team but there is a desire to have a new maintainer group take over or co-maintain a portion of the existing repositories. The process will involve opening a proposal to create a new maintainer group and outline the motivation for creating a new maintainer group and which repositories this new maintainer group would become responsible for. As part of the proposal, members of the initial maintainer group should be suggested or a request for volunteers should be open. Then the current maintainers will have to vote on whether or not they want to either cede/share control over the repositories proposed and whether or not the maintainers proposed are suitable.
+
+### New Maintainer Group for Covering Repositories that Cross Maintainer Groups
+When a project that requires coordination of components across multiple repositories that fall under the responsibility of existing maintainer groups but there are people that would like to be able to maintain the project as a whole.The process will involve opening a proposal to create a new maintainer group and outline the motivation for creating a new maintainer group and which repositories this new maintainer group would become responsible for. As part of the proposal, members of the initial maintainer group should be suggested or a request for volunteers should be open. Then the current maintainers of all repositories that are part of the proposal will have to vote on whether or not they want to share control over the repositories proposed and whether or not the maintainers proposed are suitable.
```

Comment:
```suggestion
### New Maintainer Group for Covering Repositories that Cross Maintainer Groups
When a project that requires coordination of components across multiple repositories that fall under the responsibility of existing maintainer groups but there are people that would like to be able to maintain the project as a whole.The process will involve opening a proposal to create a new maintainer group and outline the motivation for creating a new maintainer group and which repositories this new maintainer group would become responsible for. As part of the proposal, members of the initial maintainer group should be suggested or a request for volunteers should be open. Then the steering committee and current maintainers of all repositories that are part of the proposal will have to vote on whether or not they want to share control over the repositories proposed and whether or not the maintainers proposed are suitable.
```

I don't totally follow the use case here. Is this when you want to consolidate two separate groups into one group?

Same comment as before. I think steering committee should be part of all group-related decisions.

Reply:
> I don't totally follow the use case here. Is this when you want to consolidate two separate groups into one group?

This spawned out the UBI workflow. Ozzy expressed interest in becoming a maintainer of all of the components that make up the UBI workflow however those components span multiple maintainer teams. So I thought it might makes sense to make something like the UBI Maintainer team and give them joint access to UBI related repositories.

> Same comment as before. I think steering committee should be part of all group-related decisions.

I am not sure that I understand why you would want the Steering Committees approval here as it seems like something that could be hashed out among the existing maintainers. So I am curious what benefit you see to the Steering Committee being involved

Reply:
I don't personally like the idea of having groups that overlap. I think it has the potential to create confusion about ownership or introduce ways for conflicts that are difficult to resolve, like if there are two groups that are maintainers of a repo and they each have different visions about how something should be done, or if there are two groups that are maintainers and one group does not want an individual as a new member, but then the other group takes on that person as a new member. It adds complications.

I think it would also make voting calculations more complicated. We require a percentage of maintainers to vote for an issue before it's approved. Right now, that means you look at the owning team of the RFC and use the team size to calculate the required number of votes. With this proposal, it's not that simple. You need to do more assessment in terms of what teams must vote and what the required number of positive votes is to pass an RFC. Because non-votes essentially count as No votes, that could make passing RFCs harder as well.

I think we can always evaluate our group structure. If over time, it doesn't represent the ownership of repos then we can adjust things. Otherwise, I think users just need to become members of multiple groups. It is a little more work, but it allows each group to fully own membership and the owned code/repos.

Reply:
I am ok with removing this. Maybe it is something we look back on, maybe not.